syncer/streamer_controller.go

```diff
@@ -476,3 +497,33 @@ func isConnectionRefusedError(err error) bool {
 
 	return strings.Contains(err.Error(), "connect: connection refused")
 }
+
+// retryStrategy.
+type retryStrategy interface {
+	CanRetry(error) bool
+}
+
+type alwaysRetryStrategy struct{}
+
+func (s alwaysRetryStrategy) CanRetry(error) bool {
+	return true
+}
+
+// maxIntervalRetryStrategy allows retry the duration since last retry is bigger than specified interval.
```

Comment:
maybe this comment is better?

`allow retry when the retry interval is greater than the set interval.`